### ISSUE TITLE
Philips Hue App require to update the 'Emulated Hue' hub

### ISSUE THREAD

**Issue by MrGravgaard** [marcelveldt/hass_emulated_hue#8]
The Philips Hue App versio 3.37.0 (11071) keeps requesting to update the hub. Most of the features in the app is disabled until the hub is updated (which of course is not possible). 

All in all: The Emulated Hue is not functional and working sence HUE App 3.37.0 (11071)

**Comment by townsmcp** [comment]
+1 for this issue

**Comment by Hukuma1** [comment]
+1 for this I guess along with https://github.com/marcelveldt/hass_emulated_hue/issues/6 (need to update the "lights" as well lol)

**Comment by dkduck** [comment]
Nothing works when this update is comming all the time, can´t do anything, nothing works.
Please look into this issue, thanks :-)

**Comment by CiquattroFPV** [comment]
Yes I'm blocked on this too.. there is an update available and I'm not able to link lights

**Comment by Hukuma1** [comment]
@CiquattroFPV Noticed you put in a PR with new Hub version. Thank you! Could the same be done for third party lights possibly? https://github.com/marcelveldt/hass_emulated_hue/issues/6

**Comment by cmsimike** [comment]
@Hukuma1 are you asking because you can't add lights into the entertainment area without a light update?

**Comment by Hukuma1** [comment]
@cmsimike I am. Is that an easy issue to fix on my own? Didn't see any documentation on that issue. :/

**Comment by cmsimike** [comment]
I am experiencing a similar issue.
Today's my first day with this repo so trying to wrap my head around possibly a bug vs me doing stuff wrong.

**Comment by dkduck** [comment]
No lights can be added, cause of the update and when running the update, it fails every time

**Comment by CiquattroFPV** [comment]
> No lights can be added, cause of the update and when running the update, it fails every time

Ye.. I hope he will update it

**Comment by CiquattroFPV** [comment]
if you need a fix add my repository and install it again until he's not update it https://github.com/CiquattroFPV/hassio-addons/

**Comment by Hukuma1** [comment]
Thank you, @CiquattroFPV. Just installed it. However the issue of the actual light firmware needing updates is still there. Is there a way to bump the light firmware up as well? Not sure if that's as easy of a fix as the hub version.

![76688225-283b1480-65e8-11ea-8638-fcab3bd7be74](https://user-images.githubusercontent.com/54511555/80312792-3e69f200-879c-11ea-8b8f-34e9d2246ad7.png)

**Comment by CiquattroFPV** [comment]
I'm sorry but I don't know at the moment

**Comment by Ficik** [comment]
you can register the entertainment group manually by skiping the hue app and calling few api calls instead:

1. get username (link-mode needs to be active, this call will trigger notification in hass if it's not)
```
POST /api/ 
{"devicetype": "curl", "username": "curl"}
```

it returns username replace username in url in all other calls

2. figure out light ids you want to use in your group (id is key of resulting map)

```
GET /api/username/lights
```

3. create group:

```
POST /api/username/groups

{
    "class": "Other",
    "type": "Entertainment",
    "name": "My Entertainment", // rename the group
    "lights": [
        "14" // use ids from step 2
    ],
    "sensors": [],
    "action": {
        "on": true,
        "reachable": true,
        "mode": "homeautomation",
        "bri": 254,
        "colormode": "xy",
        "xy": [
            0.146,
            0.184
        ],
        "effect": "none"
    },
    "state": {
        "any_on": true,
        "all_on": false
    },
    "locations": { 
        // its map of lightId: [x,y,z] position <-1, 1> float, you can change position in hue app later
        // just add all your lights or the hue app will crash
    	"14": [0,0,0] 
    },
    "stream": {
    	"active": false,
    	"owner": null,
    	"proxynode": "/bridge",
    	"proxymode": "auto"
    }
}
```

4. you should see your entertainment group in hue app and you use it to test it and configure position of your lights

while figuring out required options of entertainment groups i had issues with updating it with put request, so instead I just deleted it with (DELETE /api/username/groups/1). If you're looking for documentation of hue bridge api found this after tons of googling : 
- https://github.com/peter-murray/node-hue-api/blob/master/docs/group.md#entertainment
- http://www.burgestrand.se/hue-api/api/auth/registration/

good luck

**Comment by Hukuma1** [comment]
@Ficik Thanks for this! I got a lot farther than before with this. :) Was able to add the lights, move them. However once I click on them it still says they are not supported. So long as I don't enter those menus it seems to still function. Going to "Test area" mode the colors do not keep up, however. And even when utilizing them using official Hue Sync Desktop app they did not seem to keep up either. I'm guessing some sort of connection hiccup that would need to be further debugged. But glad to have been able to finally add lights to test this emulation code. Thanks again! 

For anyone else wanting to try the workaround above I found RestMan Google Chrome extension to be the easiest to quickly POST/GET code for a newbie like myself.
